warehouse/query-core/src/main/java/datawave/query/jexl/DelayedIndexOnlyContext.java

```diff
@@ -0,0 +1,131 @@
+    
+    @Override
+    public Object get(String name) {
+        // only do something special if there is delayed work to do
+        if ((null != name) && delayedFields.contains(name) && !fetched.contains(name)) {
+            // fetch the field that was delayed
+            List<Document> documentFragments = null;
+            try {
+                documentFragments = fetchOnDemand(name);
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to fetch delayed index only fragments for field: " + name, e);
+            }
+            
+            // aggregate all of that into the delegate
+            for (Document document : documentFragments) {
+                document.visit(Collections.singleton(name), delegate);
+            }
+            
+            // flag that this field has now been fetched
+            fetched.add(name);
+        }
+        
+        return delegate.get(name);
+    }
+    
+    /**
+     * Use the IteratorBuildingVisitor limit to the current docRange to parse all delayed sub trees of the query. From those delayed sub trees initialize all
+     * iterators matching the target field and aggregate all partial Documents into a list
+     * 
+     * @param name
+     *            the name of the field to fetch on demand
+     * @return the list of Document objects that were fetched from all delayed iterators associated with the on-demand field
+     */
+    private List<Document> fetchOnDemand(String name) throws IOException {
+        List<Document> documentList = new ArrayList<>();
+        
+        // limit the ranges to use to the current document
+        iteratorBuildingVisitor.limit(docRange);
+        
+        // for each sub tree build the nested iterator
+        for (JexlNode subTree : delayedSubTrees) {
+            // short circuit trees that have no identifiers
+            if (JexlASTHelper.getIdentifiers(subTree).size() == 0) {
+                continue;
+            }
+            
+            // construct all index iterators for this sub tree
+            subTree.jjtAccept(iteratorBuildingVisitor, null);
+            NestedIterator<Key> subTreeNestedIterator = iteratorBuildingVisitor.root();
+            if (subTreeNestedIterator != null) {
+                // get all the leaf nodes, discarding the rest
+                Collection<NestedIterator<Key>> subTreeLeaves = subTreeNestedIterator.leaves();
+                // for each leaf, see if its a match for the target field
+                for (NestedIterator<Key> leaf : subTreeLeaves) {
+                    // grab the field off the leaf and make sure it matches
+                    if (leaf instanceof FieldedIterator && name.equals(((FieldedIterator) leaf).getField())) {
+                        // init/seek the leaf
+                        leaf.initialize();
+                        if (leaf instanceof SeekableIterator) {
+                            ((SeekableIterator) leaf).seek(docRange, null, false);
```

Comment:
This needs to be updated to pass the columnFamilies/include flag from the QueryIterator